### PR TITLE
[Storage] Fix file cmdlet context issue when current context doesn't match Track1 object context from input 

### PR DESCRIPTION
**Pull request by yifanz0**
## Description

Fix file cmdlet context issue
The issue happens when setting a current storage account first, and then call the file cmdlets with inputs of Track1 object but without an explicit context

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage.Management/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,19 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Fix file cmdlets context issue when the current context doesn't match with the context of input Track1 object
```
Comment (blueww) [inline src/Storage/Storage.Management/ChangeLog.md:21]: Track1 is internal word, should not use it in public doc.
How about: 
Fix file cmdlets potential context issue when the current context doesn't match with the credential of input Azure File object.

File: src/Storage/Storage/File/Cmdlet/StartAzureStorageFileCopy.cs [hunk 2]
```diff
@@ -251,8 +251,9 @@ public override void ExecuteCmdlet()
         {
             if(this.DestFile != null)
             {
-                // when only track1 object input, will miss storage context, so need to build storage context for prepare the output object.
-                if (this.DestContext == null)
+                // Build and set storage context for the output object when
+                // 1. input track1 object and storage context is missing 2. the current context doesn't match the context of the input object 
+                if (ShouldSetContext(this.Context, this.DestFile.ServiceClient))
```
Comment (blueww) [inline src/Storage/Storage/File/Cmdlet/StartAzureStorageFileCopy.cs:256]: should be "this.DestContext", instead of "this.Context"

Comment (yifanz0) [reply src/Storage/Storage/File/Cmdlet/StartAzureStorageFileCopy.cs]: Done

File: src/Storage/Storage.Management/ChangeLog.md (hunk 1 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/ChangeLog.md]: Yes this one definitely looks better. Done.

**Review by blueww — DISMISSED** [review]

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**